[libc] Add chown and getgid implementations

shubhe25p opened the pull request:
Implements chown and getgid per the POSIX specification and adds corresponding unit tests. getgid is added as it is required by the chown unit tests. This PR will address #165785

michaelrj-google commented on libc/src/unistd/linux/CMakeLists.txt:
```diff
@@ -25,6 +25,21 @@ add_entrypoint_object(
+  DEPENDS
+    libc.hdr.types.uid_t
+    libc.hdr.types.gid_t
+    libc.include.unistd
```
these functions don't depend on the `unistd` header so they don't need `libc.include.unistd` in their `DEPENDS`.

shubhe25p replied:
yes sure will remove it, thanks for the feedback

michaelrj-google reviewed (commented):
Overall looks good, I'll approve one you clean up the clang-format issues and address the comment

Commit:
[libc] Add chown and getgid implementations

Implements chown and getgid per the POSIX specification
and adds corresponding unit tests. getgid is added as it
is required by the chown unit tests.

michaelrj-google reviewed (approved):
LGTM, do you need me to merge this for you?

shubhe25p commented:
Yes please feel free to merge, I am not sure if I have the permissions.
